Integrations/integration-Jask.yml

```diff
@@ -0,0 +1,1003 @@
+
+
+    def translate_last_seen(last):
+        """
+        Convert last-seen argument to querystring
+        """
+        if not last or last == 'All time':
+            return ''
+        if last == 'Last week':
+            return 'timestamp:[%d TO *]' % (convertDateToUnix(datetime.utcnow()) - 7 * 24 * 60 * 60 * 1000)
+        if last == 'Last 48 hours':
+            return 'timestamp:[%d TO *]' % (convertDateToUnix(datetime.utcnow()) - 2 * 24 * 60 * 60 * 1000)
+        if last == 'Last 24 hours':
+            return 'timestamp:[%d TO *]' % (convertDateToUnix(datetime.utcnow()) - 24 * 60 * 60 * 1000)
+
+
+    def _add_list_to_q(q, translate):
+        """
+        Add arguments in the translate dictionary to querystring
+        """
+        for v in translate:
+            argList = argToList(demisto.getArg(translate[v]))
+            if len(argList) == 1:
+                q += ' AND ' + v + ':(%s)' % (argList[0])
+            elif len(argList) > 1:
+                q += ' AND ' + v + ':(%s)' % (' OR '.join(argList))
+        return q
+
+
+    def _add_time_to_q(q):
+        """
+        Add the time filter to the query string
+        Defaults to All time if no fields specified
+        """
+        lastSeen = demisto.getArg('last-seen')
+        timeFrom = demisto.getArg('time-from')
+        timeTo = demisto.getArg('time-to')
+        if lastSeen:
+            if timeFrom or timeTo:
+                return_error('You cannot specify absolute times [time-to, time-from] ' +
+                    'with relative time [last-seen]')
+            else:
+                if translate_last_seen(demisto.getArg('last-seen')) != '':
+                    q += ' AND ' + translate_last_seen(demisto.getArg('last-seen'))
+        elif timeFrom and timeTo:
+            q += ' AND timestamp:[%d TO %d]' % (convertStringDateToUnix(timeFrom), convertStringDateToUnix(timeTo))
+        elif timeFrom or timeTo:
+            return_error('You must specify both absolute times [time-to, time-from] ' +
+                'or relative time [last-seen]')
+        return q
+
+
+    def search_insights():
+        """
+        Search insights using available filters
+        """
+        q = _add_time_to_q('*')
+        q = _add_list_to_q(q, {'workflow_status':'status', 'rating': 'rating', 'group_assigned_to': 'assigned_team', 'assigned_to': 'assigned-user'})
+        QUERY['q'] = q
+        QUERY['offset'] = demisto.getArg('offset')
+        QUERY['limit'] = demisto.getArg('limit')
+        QUERY['sort_by'] = demisto.getArg('sort')
+        respJSON = req('GET', 'search/alerts')
+        insights = []
+        for insight in respJSON['objects']:
+            readable_insight = insight_to_readable(insight)
+            readable_insight['IpAddress'] = demisto.get(insight, 'asset.ip')
+            readable_insight['InsightTime'] = demisto.get(insight, 'timestamp')
+            insights.append(readable_insight)
+        ec = {'Jask.Insight': insights}
```

Comment:
`Jask.Insight` => `Jask.Insight(val.Id === obj.Id)`
You should make sure that in case the insight is already in the context then next time we update that object and not append new insight to the context